src/kv/snapshot.h

```diff
@@ -41,7 +41,7 @@ namespace kv
     std::vector<uint8_t> serialise(
       std::shared_ptr<AbstractTxEncryptor> encryptor)
     {
-      KvStoreSerialiser serialiser(encryptor, version, true);
+      KvStoreSerialiser serialiser(encryptor, version, version - 1, true);
```

Comment:
Perhaps add a short comment here to explain that the value doesn't really matter for snapshots?